singularity-eos/eos/eos_base.hpp

```diff
@@ -128,6 +128,26 @@ class EosBase {
   struct is_raw_pointer
       : std::is_same<std::remove_reference_t<std::remove_cv_t<T>>, R *> {};
 
+  // Generic evaluator
+  // JMM: These macros disable checks that the functor called by evaluate
+  // is __host__ __device__
+  // This is desirable, because you may want to call device-side batched
+  // evaluations (or inner loop dispatch with hierarchical parallelism)
+  // or you may want to call Evaluate from the host to, e.g., launch
+  // a reduction operation or somesuch.
+  // Note that this is a little dangerous as the compiler won't tell you
+  // if you messed up and ACTUALLY called a host-side function from
+  // device. The code will just segfault.
+#if defined(__CUDACC__)
+#pragma nv_exec_check_disable
```

Comment:
yeah this is a bit of a hammer IMO.

Reply:
Require that if the downstream user to place the pragmas if they are calling host functions and want warning free code. I assume if one marks the `operator()` as a portable that it won't give a warning?

You could try constexpr. 

Either way, hiding a (admittedly sometimes annoying) warning that may save a developer a lot of time does not feel like the right compromise.

Reply:
Agreed that the pragma solution is dangerous. The constexpr solution works nicely. Thanks for the suggestion @dholladay00 !